g_docs.go

```diff
@@ -211,48 +244,38 @@ func analisysNewNamespace(ce *ast.CallExpr) (first string, others []ast.Expr) {
 	return
 }
 
-func analisysNSInclude(baseurl string, ce *ast.CallExpr) string {
+func analisysNSInclude(baseURL string, ce *ast.CallExpr) string {
 	cname := ""
 	for _, p := range ce.Args {
 		x := p.(*ast.UnaryExpr).X.(*ast.CompositeLit).Type.(*ast.SelectorExpr)
 		if v, ok := importlist[fmt.Sprint(x.X)]; ok {
 			cname = v + x.Sel.Name
 		}
 		if apis, ok := controllerList[cname]; ok {
-			for rt, item := range apis {
-				tag := ""
-				if baseurl != "" {
-					rt = baseurl + rt
-					tag = strings.Trim(baseurl, "/")
-				} else {
-					tag = cname
-				}
-				if item.Get != nil {
-					item.Get.Tags = append(item.Get.Tags, tag)
+			for route, item := range apis {
+				tag := cname
+				if baseURL != "" {
+					route = baseURL + route
+					tag = strings.Trim(baseURL, "/")
 				}
-				if item.Post != nil {
-					item.Post.Tags = append(item.Post.Tags, tag)
-				}
-				if item.Put != nil {
-					item.Put.Tags = append(item.Put.Tags, tag)
-				}
-				if item.Patch != nil {
-					item.Patch.Tags = append(item.Patch.Tags, tag)
-				}
-				if item.Head != nil {
-					item.Head.Tags = append(item.Head.Tags, tag)
-				}
-				if item.Delete != nil {
-					item.Delete.Tags = append(item.Delete.Tags, tag)
-				}
-				if item.Options != nil {
-					item.Options.Tags = append(item.Options.Tags, tag)
+
+				// replicate item to make referenced value untouched,
+				// because if the endpoint shares the same controller with another
+				// endpoints the referenced value will be used again and the new
+				// endpoint will need the original value rather than the edited value.
+				replicatedItem, err := replicateSwaggerItem(item)
+				if err != nil {
+					ColorLog("[ERRO] failed replicating swagger item, got: %s\n", err)
+					os.Exit(1)
 				}
+
+				enrichItem(replicatedItem, tag, route)
```

Comment:
Wondering if this is really necessary? 🤔 Can we just pass a non-pointer to enrich and return a new, enriched swagger-item?

Also I quite dont understand the comment above. Does it mean two endpoints might share the same controller?